### PR TITLE
xt: create show page for each image

### PR DESCRIPTION
**Pull request by Xin-Tong**
Add a show page to each image.

https://image-sharer-xin-stage-pr-30.herokuapp.com/

### CODE REVIEW

File: app/controllers/images_controller.rb [hunk 1]
```diff
@@ -33,6 +33,14 @@ def destroy
     redirect_to images_path
   end
 
+  def show
+    @image = Image.find_by(id: params[:id])
+    unless @image
+      flash[:danger] = 'Cannot find this image'
+      redirect_to images_path
+    end
```
Comment (rmacklin) [inline app/controllers/images_controller.rb]: I think it'd actually be better to just let the Rails default 404 handling happen here. This action would just be: `@image = Image.find(params[:id])`

File: app/views/images/index.html.erb [hunk 2]
```diff
@@ -1,21 +1,7 @@
...
-            <% image.tag_list.each do |tag| %>
-              <%= link_to images_path(tag: tag) do %>
-                <span class="label label-info"><%= tag %></span>
-              <% end %>
-            <% end %>
-          </span>
-          <%= link_to 'Delete', image_path(image), method: :delete, data: { confirm: 'Are you sure?' }, class: 'btn btn-secondary' %>
-        </div>
-      </div>
+      <%= render partial: image %>
```
Comment (rmacklin) [inline app/views/images/index.html.erb]: This should just be `render: image`

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -102,4 +102,22 @@ class ImagesControllerTest < ActionController::TestCase
...
+    assert_response :success
+    assert_select "img:match('src',?)", image_url
+  end
+
+  test 'show an invalid image' do
+    get :show, id: -1
+
+    assert_redirected_to images_path
+    assert_equal 'Cannot find this image', flash[:danger]
+  end
```
Comment (rmacklin) [inline test/controllers/images_controller_test.rb]: Remove this test after refactoring the action.

File: app/controllers/images_controller.rb [hunk 4]
```diff
@@ -33,6 +33,14 @@ def destroy
     redirect_to images_path
   end
 
+  def show
```
Comment (rmacklin) [inline app/controllers/images_controller.rb]: Now that we have a show action, can you make `create` redirect to it? Also, can you make the images on the index page link to the corresponding show page?

File: test/flow_tests/images_flow_test.rb [hunk 5]
```diff
@@ -102,4 +102,15 @@ class ImageFlowTest < FlowTestCase
     page.assert_current_path images_path
     assert page.has_text?('Cannot find this image')
   end
+
+  test 'show an image' do
```
Comment (rmacklin) [inline test/flow_tests/images_flow_test.rb]: After addressing https://github.com/AppFolioOnboarding/image-sharer-xin/pull/30#discussion_r61203705, I think you can just incorporate the show page into existing flow tests.

File: app/controllers/images_controller.rb (hunk 4 above)
Comment (Xin-Tong) [inline app/controllers/images_controller.rb]: Of course, can I make `create` redirect to it in the PR? 
By "link to the corresponding show page", did you mean this? https://github.com/AppFolioOnboarding/image-sharer-xin/pull/30/files#diff-d8dfe982a08739156d9e80f722bb4b2eR2

File: app/controllers/images_controller.rb (hunk 1 above)
Comment (Xin-Tong) [inline app/controllers/images_controller.rb]: Sounds reasonable. I will do it.

File: app/views/images/index.html.erb (hunk 2 above)
Comment (Xin-Tong) [inline app/views/images/index.html.erb]: 👍

File: test/controllers/images_controller_test.rb (hunk 3 above)
Comment (Xin-Tong) [inline test/controllers/images_controller_test.rb]: 👍

File: app/controllers/images_controller.rb (hunk 4 above)
Comment (rmacklin) [inline app/controllers/images_controller.rb]: Yes, it should be in this PR. And yes, that link is good. Would be good to exercise that link in a flow test.

Comment (Xin-Tong) [inline app/controllers/images_controller.rb]: I don't know how to click on an image. So I just tested the link like this https://github.com/AppFolioOnboarding/image-sharer-xin/blob/ShowPage/test/flow_tests/images_flow_test.rb#L112

**Pull request merged**